File: application/core/src/main/java/br/com/beblue/bluerecords/core/usecase/DiscoUseCaseImpl.java
package br.com.beblue.bluerecords.core.usecase;

import br.com.beblue.bluerecords.core.command.CadastraDiscoCommand;
import br.com.beblue.bluerecords.core.command.ConsultaDiscoCommand;
import br.com.beblue.bluerecords.core.entitidade.Disco;
import br.com.beblue.bluerecords.core.entitidade.Genero;
import br.com.beblue.bluerecords.core.paginacao.Paginacao;
import br.com.beblue.bluerecords.core.repositorio.DiscoRepositorio;
import br.com.beblue.bluerecords.core.repositorio.DiscoSpotifyClient;

import java.util.List;

public class DiscoUseCaseImpl implements DiscoUseCase {

    private DiscoRepositorio discoRepositorio;
    private DiscoSpotifyClient discoSpotifyClient;

    public DiscoUseCaseImpl(DiscoRepositorio discoRepositorio ,
                            DiscoSpotifyClient discoSpotifyClient) {
        this.discoRepositorio = discoRepositorio;
        this.discoSpotifyClient = discoSpotifyClient;
    }

    @Override
    public Disco consultar(Integer id)
    {
        return discoRepositorio.consultar(id);
    }

    @Override
    public Boolean discosForamCadastrados() {
        return discoRepositorio.existeDiscoCadastrado();
    }

    @Override
    public List<CadastraDiscoCommand> consultarDiscosParaCadastroPorGenero(Genero genero) {
        return discoSpotifyClient.buscarDiscos(genero, 50,0);
    }

    @Override
    public Paginacao<Disco> consultarDiscos(ConsultaDiscoCommand command)
    {
        return discoRepositorio.consultarDiscos(command);
    }

    @Override
    public Disco cadastrar(CadastraDiscoCommand command) {
        return discoRepositorio.cadastrar(command);
    }

}

File: application/core/src/main/kotlin/br/com/beblue/bluerecords/core/usecase/disco/DiscoUseCaseImpl.kt
package br.com.beblue.bluerecords.core.usecase.disco

import br.com.beblue.bluerecords.core.entity.Disco
import br.com.beblue.bluerecords.core.entity.Genero
import br.com.beblue.bluerecords.core.repositorio.DiscoRepositorio
import br.com.beblue.bluerecords.core.repositorio.DiscoSpotifyClient
import br.com.beblue.bluerecords.core.usecase.disco.command.CadastrarDiscoCommand
import br.com.beblue.bluerecords.core.usecase.disco.command.ConsultaDiscosCommand

class DiscoUseCaseImpl(private val discoRepositorio: DiscoRepositorio,
                       private val discoSpotifyClient: DiscoSpotifyClient
) : DiscoUseCase
{

    override fun consultarDiscoPorId(id: Int): Disco {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun discosForamCadastrados(): Boolean {
        return discoRepositorio.existeDiscoCadastrado()
    }

    override fun consultarDiscosParaCadastroPorGenero(genero: Genero): List<CadastrarDiscoCommand>
    {
        return discoSpotifyClient.buscarDiscos(genero, 50,0)
    }

    override fun cadastrarDisco(command: CadastrarDiscoCommand): Disco {
        return discoRepositorio.cadastrar(command)
    }

}